homeassistant/components/google_assistant/trait.py

```diff
@@ -327,6 +331,38 @@ async def execute(self, command, data, params, challenge):
         }
 
 
+@register_trait
+class ObjectDetection(_Trait):
+    """Trait to object detection.
+
+    https://developers.google.com/actions/smarthome/traits/objectdetection
+    """
+
+    name = TRAIT_OBJECTDETECTION
+    commands = []
+
+    @staticmethod
+    def supported(domain, features, device_class, _) -> bool:
+        """Test if state is supported."""
+        return domain in (event.DOMAIN,)
+
+    def sync_attributes(self):
+        """Return ObjectDetection attributes for a sync request."""
+        return {}
+
+    def query_attributes(self):
```

Comment:
Shouldnt we separate this into a new function gor notifications. The code that extracts this later looks rather ugly.

Reply:
Sure, it is just a proof of concept yet, but GA does not trigger, when an event is sent

Reply:
Mode the code to a new function